snuba/web/rpc/v1/endpoint_get_traces.py

```diff
@@ -0,0 +1,418 @@
+import uuid
+from collections import defaultdict
+from typing import Any, Callable, Dict, Iterable, Sequence, Type
+
+from google.protobuf.internal.containers import RepeatedCompositeFieldContainer
+from google.protobuf.json_format import MessageToDict
+from sentry_protos.snuba.v1.endpoint_get_traces_pb2 import (
+    GetTracesRequest,
+    GetTracesResponse,
+    TraceAttribute,
+)
+from sentry_protos.snuba.v1.request_common_pb2 import PageToken, TraceItemName
+from sentry_protos.snuba.v1.trace_item_attribute_pb2 import AttributeKey, AttributeValue
+from sentry_protos.snuba.v1.trace_item_filter_pb2 import TraceItemFilter
+
+from snuba.attribution.appid import AppID
+from snuba.attribution.attribution_info import AttributionInfo
+from snuba.datasets.entities.entity_key import EntityKey
+from snuba.datasets.entities.factory import get_entity
+from snuba.datasets.pluggable_dataset import PluggableDataset
+from snuba.query import LimitBy, OrderBy, OrderByDirection, SelectedExpression
+from snuba.query.data_source.simple import Entity
+from snuba.query.dsl import Functions as f
+from snuba.query.dsl import and_cond, column, in_cond, literal, literals_array
+from snuba.query.expressions import Expression
+from snuba.query.logical import Query
+from snuba.query.query_settings import HTTPQuerySettings
+from snuba.request import Request as SnubaRequest
+from snuba.web.query import run_query
+from snuba.web.rpc import RPCEndpoint
+from snuba.web.rpc.common.common import (
+    base_conditions_and,
+    project_id_and_org_conditions,
+    timestamp_in_range_condition,
+    trace_item_filters_to_expression,
+    treeify_or_and_conditions,
+)
+from snuba.web.rpc.common.debug_info import (
+    extract_response_meta,
+    setup_trace_query_settings,
+)
+from snuba.web.rpc.common.exceptions import BadSnubaRPCRequestException
+
+_DEFAULT_ROW_LIMIT = 10_000
+
+_ATTRIBUTES: dict[
+    TraceAttribute.Key.ValueType,
+    tuple[str, AttributeKey.Type.ValueType],
```

Comment:
Nit: a named tuple/data class/typed dict would be a little cleaner here so you don't have to access it by index